### ISSUE TITLE
Username is case sensitive

### ISSUE THREAD

**Issue by aaronomnis1** [feentech/leroy#33]
Username is case sensitive.

**Comment by kiernfeeney** [comment]
Laravel doesn't provide a case insensitive database data type by default. Postgres does provide [citext](https://www.postgresql.org/docs/13/citext.html) but Laravel doesn't provide a native way to interact with it. That means baring manually altering the postgres database which isn't a good idea as migrations could override the changes. The fix I came up with is forcing all usernames to lowercase i.e. `AaronOmnis1` becomes `aaronomnis1`.

**Comment by kiernfeeney** [comment]
UPDATE:

From with the `create_users_table` database migration I added the following code

```
DB::statement('
    ALTER TABLE users 
        ALTER COLUMN username TYPE CITEXT,
        ALTER COLUMN email        TYPE CITEXT;
');
```
After the database table is created, the username and email columns are changed to CITEXT which enforces case insensitivity on the database level. I tested this at login and it seemed to work. Keep in mind that you'll need to make sure the CITEXT extension is loaded into Postgres (by default it isn't).
